app/javascript/mastodon/reducers/relationships.js

```diff
@@ -23,6 +23,14 @@ const normalizeRelationships = (state, relationships) => {
   return state;
 };
 
+const setDomainBlocking = (state, accounts, blocking) => {
+  accounts.forEach(id => {
+    state = state.setIn([id, 'domain_blocking'], blocking);
```

Comment:
What about using `withMutations`?

Reply:
Hm, I don't know `withMutations` and I'm not sure how it would make things better?

Reply:
`withMutations` makes an immutable object mutable in a given callback, and returns the muted object. It improves performance by removing redundant immutable object construction. And you won't have to write something like `state = state.setIn(...);`. Just `state.setIn(...);` will be fine.